Update to obsproc.v1.0.2 and prepobs.v1.0.1

KateFriedman-NOAA opened the pull request:
**Description**

This PR updates the `develop` branch to the latest obsproc and prepobs packages that resulted from the WCOSS2 port (GFSv16): `obsproc.v1.0.2` and `prepobs.v1.0.1`. This PR also includes some prep job cleanup and fixes.

Resolves #1033

Changes:

1. Add `launcher_PREP` variable to `HERA.env` and `ORION.env`; equals "srun" on those platforms, already set in `WCOSS2.env`.
2. Update `jobs/rocoto/prep.sh` to invoke new obsproc JJOB: replace `$HOMEobsproc_network/jobs/JGLOBAL_PREP` with `$HOMEobsproc/jobs/JOBSPROC_GLOBAL_PREP`
3. Add prepobs module load to `module_base.hera.lua` and `module_base.orion.lua` (already in `module_base.wcoss2.lua`); the prepobs package is now accessed with a module load.
4. Update `HOMEobsproc` variables in `config.base.emc.dyn` and `config.base.nco.static`: replace `HOMEobsproc_prep`, `HOMEobsproc_network` and `HOMEobsproc_global` with `HOMEobsproc`.
5. Update `config.prep` to add needed `cdate10`, `COMPONENT`, and `COMINtcvital` variables.
6. Retire prepbufr config (`config.prepbufr`) and reference in `config.resources` prep section.

**Type of change**

- [x New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [x] Cycled tests on Hera, Orion, WCOSS2

Testing on Hera and Orion reproduced control tests (identical prepbufr files).
  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

KateFriedman-NOAA committed:
Retire config.prepbufr

- The contents of config.prepbufr are only setting GESROOT;
which is no longer used in the GFS in operations.
- Don't need to set GESROOT so don't need config.prepbufr.
- Delete config.prepbufr.
- Remove "prepbufr" from config list in prep.sh.

Refs #1033

KateFriedman-NOAA committed:
Update primary obsproc JJOB call in prep.sh

The updated obsproc package renamed the primary global JJOB script
and the HOMEobsproc variable is the only home path for the package,
so update prep.sh to now call:
$HOMEobsproc/jobs/JOBSPROC_GLOBAL_PREP
instead of the old:
$HOMEobsproc_network/jobs/JGLOBAL_PREP

Refs #1033

KateFriedman-NOAA committed:
Add prepobs module load to R&D module_base modulefiles

- Prepend MODULEPATH with local install of prepobs package on R&Ds.
- Load prepobs/1.0.1 module from local install.

The prepobs package is now a module that is loaded at runtime.

Refs #1033

KateFriedman-NOAA committed:
Update HOMEobsproc paths in config.base

- Consolidate HOMEobsproc_prep, HOMEobsproc_network, and HOMEobsproc_global
into a single HOMEobsproc path variable.
- Set HOMEobsproc to local install location and current version 1.0.2 in
config.base.emc.dyn.
- Set HOMEobsproc to operational package location and current version 1.0.2
in config.base.nco.static.

Refs #1033

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-obsproc

KateFriedman-NOAA committed:
Add cdate10 setting to config.prep

The cdate10 variable is ${PDY}${cyc} and is used by the obsproc package
to set CDATE10/CDATEMM.

Refs #1033

KateFriedman-NOAA committed:
Add launcher_PREP to HERA and ORION env files

Refs #1033

KateFriedman-NOAA committed:
Retire prepbufr step from config.resources

Refs #1033

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-obsproc

KateFriedman-NOAA committed:
Add needed COMINtcvital path to config.prep

- The prep job needs to know COMINtcvital so it can find the syndata.tcvitals file
from the global dump archive in developer testing; uses compath.py in ops.
- Add COMPONENT setting for use in COMINtcvital setting.

Refs #1033

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-obsproc

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-obsproc

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-obsproc

aerorahul commented on parm/config/config.prep:
```diff
@@ -17,6 +18,8 @@ export DO_RELOCATE="NO"
 export TROPCYQCRELOSH="$HOMEgfs/scripts/exglobal_atmos_tropcy_qc_reloc.sh"
 export SENDCOM=YES
 
+export COMPONENT=${COMPONENT:-atmos}
```
Is this really something that will be set from above?
Isn't component here "always" `atmos`?

aerorahul commented on parm/config/config.prep:
```diff
@@ -9,6 +9,7 @@ echo "BEGIN: config.prep"
 . $EXPDIR/config.resources prep
 
 export DO_MAKEPREPBUFR="YES"   # if NO, will copy prepbufr from globaldump
+export cdate10=${PDY}${cyc}
```
Who is needing this variable to be set in their env.?
Is production setting this in `ecf` of the obsprocs?
Just curious.

KateFriedman-NOAA replied:
We have to set `cdate10` for obsproc `scripts/exglobal_makeprepbufr.sh`:

```
cdate10=${cdate10:-`cut -c7-16 ncepdate`}
```

https://github.com/NOAA-EMC/obsproc/blob/develop/scripts/exglobal_makeprepbufr.sh#L78

I had them add the override this year so we can pass `cdate10` in for development runs. In ops the default (`ncepdate`) is used. The same update was made in the GFSv16 system.

aerorahul reviewed (approved):
looks good.
just a couple of minor comments.

KateFriedman-NOAA committed:
Remove COMPONENT use from config.prep

- Remove COMPONENT setting.
- Remove use of COMPONENT in setting COMINtcvital and hardcode "atmos" in path

Refs #1033

aerorahul reviewed (approved):
looks good.
thanks~
